ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/Utils.java

```diff
@@ -74,4 +95,43 @@ public static void writeOutput(List<String> values, Path outputFile, HiveConf hi
       return db.getTablesByPattern(dbName, tblPattern);
     }
   }
+
+  public static void setDbBootstrapDumpState(Hive hiveDb, String dbName, String dumpState) throws HiveException {
+    Database database = hiveDb.getDatabase(dbName);
+    if (database == null) {
+      return;
+    }
+
+    Map<String, String> newParams = new HashMap<>();
+    newParams.put(BOOTSTRAP_DUMP_STATE_KEY, dumpState);
+    Map<String, String> params = database.getParameters();
+
+    // if both old params are not null, merge them
+    if (params != null) {
+      params.putAll(newParams);
+      database.setParameters(params);
+    } else {
+      // if one of them is null, replace the old params with the new one
+      database.setParameters(newParams);
+    }
+
+    hiveDb.alterDatabase(dbName, database);
+    return;
+  }
+
+  private static String getDbBootstrapDumpStateFromParameters(Map<String, String> params) {
+    if ((params != null) && (params.containsKey(BOOTSTRAP_DUMP_STATE_KEY))){
+      return params.get(BOOTSTRAP_DUMP_STATE_KEY);
+    }
+    return null;
+  }
+
+  public static boolean isBootstrapDumpInProgress(Hive hiveDb, String dbName) throws HiveException {
+    Database database = hiveDb.getDatabase(dbName);
+    if (database == null) {
+      return false;
+    }
+    String dumpState = getDbBootstrapDumpStateFromParameters(database.getParameters());
+    return ((dumpState != null) && dumpState.equals(ReplDumpState.ACTIVE.toString()));
+  }
```

Comment:
getDbBootstrapDumpStateFromParameters and isBootstrapDumpInProgress can be simplified preventing non negative checks, with something like 


  public static boolean isBootstrapDumpInProgress(Hive hiveDb, String dbName) throws HiveException {
    Database database = hiveDb.getDatabase(dbName);
    if (database == null) {
      return false;
    }
    Map<String, String> parameters = database.getParameters();
    if (parameters == null) {
      return false;
    }
    return parameters.containsKey(BOOTSTRAP_DUMP_STATE_KEY) && parameters
        .get(BOOTSTRAP_DUMP_STATE_KEY).equals(ReplDumpState.ACTIVE.toString());
  }